runtime/common/src/main/java/edu/snu/nemo/runtime/common/optimizer/pass/runtime/DataSkewRuntimePass.java

```diff
@@ -75,52 +81,75 @@ public PhysicalPlan apply(final PhysicalPlan originalPlan,
         .collect(Collectors.toList());
 
     // Get number of evaluators of the next stage (number of blocks).
-    final Integer taskListSize = optimizationEdges.stream().findFirst().orElseThrow(() ->
+    final Integer reducerTaskNum = optimizationEdges.stream().findFirst().orElseThrow(() ->
         new RuntimeException("optimization edges are empty")).getDst().getTaskIds().size();
 
     // Calculate keyRanges.
-    final List<KeyRange> keyRanges = calculateHashRanges(metricData.right(), taskListSize);
+    final List<KeyRange> keyRanges = calculateHashRanges(metricData.right(), reducerTaskNum);
 
     // Overwrite the previously assigned hash value range in the physical DAG with the new range.
     optimizationEdges.forEach(optimizationEdge -> {
       // Update the information.
-      final List<KeyRange> taskIdxToHashRange = new ArrayList<>();
-      IntStream.range(0, taskListSize).forEach(i -> taskIdxToHashRange.add(keyRanges.get(i)));
+      final Map<Integer, KeyRange> taskIdxToHashRange = new HashMap<>();
+      for (int taskIdx = 0; taskIdx < reducerTaskNum; taskIdx++) {
+        taskIdxToHashRange.put(taskIdx, keyRanges.get(taskIdx));
+      }
       optimizationEdge.setTaskIdxToKeyRange(taskIdxToHashRange);
     });
 
     return new PhysicalPlan(originalPlan.getId(), physicalDAGBuilder.build());
   }
 
+  private boolean containsSkewedHash(final List<Integer> skewedHashes,
+                                     final int startingHash, final int finishingHash) {
+    for (int h = startingHash; h < finishingHash; h++) {
+      if (skewedHashes.contains(h)) {
+        return true;
+      }
+    }
+    return false;
+  }
+
   /**
    * Method for calculating key ranges to evenly distribute the skewed metric data.
    *
    * @param aggregatedMetricData the metric data.
-   * @param taskListSize the size of the task list.
+   * @param taskNum the size of the task list.
    * @return the list of key ranges calculated.
    */
   @VisibleForTesting
   public List<KeyRange> calculateHashRanges(final Map<Integer, Long> aggregatedMetricData,
-                                            final Integer taskListSize) {
+                                            final Integer taskNum) {
     // NOTE: aggregatedMetricDataMap is made up of a map of (hash value, blockSize).
     // Get the max hash value.
```

Comment:
Can you clarify what 'max hash value' means?
Does this imply actual data sizes?